tests/__init__.py

```diff
@@ -122,7 +122,14 @@ def simulate(self, client, id, data):
         response = client.get(f'/devices/{id}')
         assert response.status_code == 200
 
-        input_nodes = response.json().get('input_nodes', {})
+        sleep(0.5)
```

Comment:
This will add delay when creating devices. Most of the tests don't need the device to have input nodes, so they don't need any delay. As this is just a test suite I thought it was acceptable to do it like this, because adding the delay in the device creation will add even more overhead.